src/test/java/org/jabref/model/entry/AuthorListTest.java

```diff
@@ -626,6 +626,25 @@ public void parseNameWithHyphenInLastName() throws Exception {
     }
 
     @Test
+    public void parseNameWithHyphenInLastNameWithInitials() throws Exception {
+        Author expected = new Author("E. S.", "E. S.", null, "El-{M}allah", null);
+        Assert.assertEquals(new AuthorList(expected), AuthorList.parse("E. S. El-{M}allah"));
+    }
+
+    @Test
+    public void parseNameWithHyphenInLastNameWithEscaped() throws Exception {
+        Author expected = new Author("E. S.", "E. S.", null, "{K}ent-{B}oswell", null);
+        Assert.assertEquals(new AuthorList(expected), AuthorList.parse("E. S. {K}ent-{B}oswell"));
+    }
+
+    @Test
+    public void parseNameWithHyphenInLastNameWhenLastNameGivenFirst() throws Exception {
+        // TODO: Fix abbreviation to be "A."
```

Comment:
Can you get this working too?

Reply:
Sadly no. I tried to fix it but the extraction of the abbreviation from the text is hidden in the whole tokenizing process and thus not easy to change.